spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTable.java

```diff
@@ -341,4 +343,42 @@ public void testDowngradeTableToFormatV1ThroughTablePropertyFails() {
         "Cannot downgrade v2 table to v1",
         () -> sql("ALTER TABLE %s SET TBLPROPERTIES ('format-version'='1')", tableName));
   }
+
+  @Test
+  public void testCatalogSpecificWarehouseLocation() throws Exception {
+    File warehouseLocation = temp.newFolder(catalogName);
+    Assert.assertTrue(warehouseLocation.delete());
+    String location = "file:" + warehouseLocation.getAbsolutePath();
+    spark.conf().set("spark.sql.catalog." + catalogName + ".warehouse", location);
+    String testNameSpace =
+        (catalogName.equals("spark_catalog") ? "" : catalogName + ".") + "default1";
+    String testTableName = testNameSpace + ".table";
+    TableIdentifier newTableIdentifier = TableIdentifier.of("default1", "table");
+
+    sql("CREATE NAMESPACE IF NOT EXISTS %s", testNameSpace);
+    sql("CREATE TABLE %s " + "(id BIGINT NOT NULL, data STRING) " + "USING iceberg", testTableName);
+
+    Table table = Spark3Util.loadIcebergCatalog(spark, catalogName).loadTable(newTableIdentifier);
+    if ("spark_catalog".equals(catalogName)) {
+      Assert.assertTrue(table.location().startsWith(spark.sqlContext().conf().warehousePath()));
```

Comment:
SparkInnerSessionCatalog will awalys use `spark.sql.warehouse.dir` as create table locations.
<img width="1212" alt="image" src="https://user-images.githubusercontent.com/36697727/227761770-7d6a00f2-4986-4a50-9920-3726dc1f9949.png">

Reply:
So two things here
1. I don't follow why "spark_catalog" should be any different here since we are still using our SparkSessionCatalog class.
2. We need to make sure that the prefix and suffix are correct in the path. It should start with our specified location and end with our db + tb

Reply:
thanks for review.
1. i found SparkSessionCatalog will away use `spark.sql.warehouse.dir`,  so I mistakenly thought that SparkSesssionCatalog was only implemented using SparkInnerHive even for the iceberg table, and now the behavior in the code is indeed the same, which confuses me,  So it seems that this is not in line with expectations, what do you think is the reason for this happening?
![image](https://user-images.githubusercontent.com/36697727/232000160-551ab282-55f7-46eb-bb36-c91bb46bbdca.png)

2. I will change the judgment of table location, thank you